SQL_Queries.txt

```diff
@@ -0,0 +1,1670 @@
+output:
++------+--------+---------+----------+
+| snum | sname  | amt     | odate    |
++------+--------+---------+----------+
+| 1002 | Serres | 5160.45 | 10/03/90 |
+| 1001 | Peel   | 4723.00 | 10/05/90 |
+| 1001 | Peel   | 9891.88 | 10/06/90 |
++------+--------+---------+----------+
+3 rows in set (1.02 sec)
+
+21) List all the largest orders for October 3, for each salesperson.
+
+SELECT snum ,MAX(amt) FROM orders WHERE odate = "10/03/90" GROUP BY snum ;
+output:
++------+----------+
+| snum | max(amt) |
++------+----------+
+| 1001 |   767.19 |
+| 1002 |  5160.45 |
+| 1004 |  1900.10 |
+| 1007 |  1098.16 |
++------+----------+
+4 rows in set (0.00 sec)
+
+22) Find all customers located in cities where Serres has customers. 
+
+SELECT cnum , cname , city FROM customers WHERE city IN (SELECT city FROM customers WHERE snum = (SELECT snum FROM salespeople WHERE sname = "serres")) ;
+output: 
++------+----------+---------+
+| cnum | cname    | city    |
++------+----------+---------+
+| 2003 | Liu      | Sanjose |
+| 2004 | Grass    | Berlin  |
+| 2008 | Cisneros | Sanjose |
++------+----------+---------+
+
+23) Select all customers with a rating above 200.
+
+SELECT cnum , cname , city , rating FROM customers WHERE rating > 200 ;
+output:
++------+----------+---------+--------+
+| cnum | cname    | city    | rating |
++------+----------+---------+--------+
+| 2004 | Grass    | Berlin  |    300 |
+| 2008 | Cisneros | Sanjose |    300 |
++------+----------+---------+--------+
+
+24) Count the number of salespeople currently having orders in the orders table. 
+SELECT COUNT(DISTINCT(snum)) AS Salespeople_Orders FROM orders ;
++--------------------+
+| Salespeople_Orders |
++--------------------+
+|                  5 |
++--------------------+
+
+25) Write a query that produces all customers serviced by salespeople with a commission above 12%. Output the customer’s name, salesperson’s name and the salesperson’s rate of commission. 
+
+SELECT c.cname , s.sname , s.comm FROM customers as c
+INNER JOIN
+salespeople AS s WHERE s.snum = c.snum AND s.comm>12 ;
+
+output:
++----------+--------+------+
+| cname    | sname  | comm |
++----------+--------+------+
+| Liu      | Serres |   13 |
+| Grass    | Serres |   13 |
+| Cisneros | RifKin |   15 |
++----------+--------+------+
+
+26) Find salespeople who have multiple customers. 
+
+SELECT s.snum , s.sname , COUNT(c.snum) AS count FROM salespeople AS s
+INNER JOIN
+customers AS c WHERE c.snum = s.snum GROUP BY c.snum HAVING count>1 ;
+output:
++------+--------+-------+
+| snum | sname  | count |
++------+--------+-------+
+| 1001 | Peel   |     2 |
+| 1002 | Serres |     2 |
++------+--------+-------+
+
+27) Find salespeople with customers located in their own cities.
+
+SELECT s.sname , c.cname , s.city FROM salespeople AS s
+INNER JOIN
```

Comment:
Not used best practice to use joins , where is 'On' clause , why not used 'On' ?